x-pack/plugins/observability_solution/observability_ai_assistant/server/plugin.ts

```diff
@@ -154,6 +162,45 @@ export class ObservabilityAIAssistantPlugin
 
     addLensDocsToKb({ service, logger: this.logger.get('kb').get('lens') });
 
+    const initResources = async (
+      request: KibanaRequest
+    ): Promise<ObservabilityAIAssistantRouteHandlerResources> => {
+      const [coreStart, pluginsStart] = await core.getStartServices();
+      const license = await firstValueFrom(pluginsStart.licensing.license$);
+      const savedObjectsClient = coreStart.savedObjects.getScopedClient(request);
+      return {
+        request,
+        service,
+        context: {
+          rac: routeHandlerPlugins.ruleRegistry.start().then((startContract) => {
+            return {
+              getAlertsClient() {
+                return startContract.getRacClientWithRequest(request);
+              },
+            };
+          }),
+          core: {
+            coreStart,
+            elasticsearch: {
+              client: coreStart.elasticsearch.client.asScoped(request),
+            },
+            uiSettings: {
+              client: coreStart.uiSettings.asScopedToClient(savedObjectsClient),
+            },
+            savedObjects: {
+              client: savedObjectsClient,
+            },
+          },
+          licensing: { license },
+        } as unknown as ObservabilityAIAssistantRequestHandlerContext,
+        logger: this.logger.get('connector'),
+        plugins: routeHandlerPlugins,
+      };
+    };
+
+    plugins.actions.registerType(getObsAIAssistantConnectorType(initResources));
```

Comment:
In case you want to support different sub-actions (different execution flows) for your connector it may be better to switch to the sub-actions framework. It is optinal but it will be difficult to switch in the future as the expected params schema is different (`{ subAction: "run", subActionsParams: { // myParams}`).

Reply:
Discussed offline and we'll implement sub action framework as a follow up given the timeline. I'll create an issue